nkms/characters.py

```diff
@@ -379,12 +378,12 @@ def grant(self, bob, uri, networky_stuff,
         # by trying differet
         # deposits and expirations on a limited number of Ursulas.
         # Users may decide to inject some market strategies here.
-        found_ursulas = policy.find_ursulas(networky_stuff, deposit,
+        found_ursulas = policy.find_ursulas(self.network_middleware, deposit,
                                             expiration, num_ursulas=n)
         policy.match_kfrags_to_found_ursulas(found_ursulas)
         # REST call happens here, as does population of TreasureMap.
-        policy.enact(networky_stuff)
-        policy.publish_treasure_map(networky_stuff)
+        policy.enact(self.network_middleware)
```

Comment:
👍

Reply:
Yeah, rolls nicely right?  I think we're probably go to adopt this everywhere.